packages/styles/scss/components/ui-shell/side-nav/_side-nav.scss

```diff
@@ -87,6 +87,7 @@
 
   .#{$prefix}--side-nav__overlay-active {
     @include breakpoint-down('lg') {
+      z-index: z('header');
```

Comment:
could we use `overlay` here instead of `header`? that seemed to work for me in the test story 
<img width="1044" alt="image" src="https://user-images.githubusercontent.com/40550942/235259610-09510346-26b7-4c1c-af74-ddf62d823fc2.png">
```suggestion
      z-index: z('overlay');
```